src/dynamics/models/elasticity_corotated_linear.rs

```diff
@@ -17,6 +17,21 @@ impl ConstitutiveModel for CorotatedLinearElasticity {
         )
     }
 
+    // https://www.math.ucla.edu/~cffjiang/research/mpmcourse/mpmcourse.pdf#subsection.6.3
+    fn elastic_energy_density(&self, deformation_gradient: Matrix<Real>) -> Real {
+        let singular_values = deformation_gradient
+            .svd_unordered(false, false)
+            .singular_values;
+        let determinant: Real = singular_values.iter().product();
+
+        self.mu
+            * singular_values
+                .iter()
+                .map(|sigma| (sigma - 1.).powi(2))
+                .sum::<Real>()
+            + self.lambda / 2. * (determinant - 1.).powi(2)
```

Comment:
I would recommend taking the whole `&Particle` as argument, for consistency with `pos_energy` and to take hardening into account the same way as in `pos_energy`.

Can be in another PR.